src/components/DayPickerRangeController.jsx

```diff
@@ -163,7 +163,7 @@ export default class DayPickerRangeController extends React.Component {
     if (focusedInput !== END_DATE) return false;
 
     if (startDate) {
-      const dayDiff = day.diff(startDate, 'days');
+      const dayDiff = day.day() - startDate.day();
```

Comment:
This is not going to work over week borders as it will only return 0-6 and `.date()` and `.dayOfYear()` have similar problems over month and year boundaries respectively. I think unfortunately, something like I recommended earlier with comparing the `startOf` both days seems the most reasonable. Alternatively, because you are guaranteed that `day` will always be at noon, you could set `startDate.hour('12)` and compare those, although that is more brittle.